src/palimpzest/dataclasses.py

```diff
@@ -233,11 +233,17 @@ def add_record_op_stats(
             self.total_op_time += record_op_stats.time_per_record
             self.total_op_cost += record_op_stats.cost_per_record
 
-    def __iadd__(self, op_stats: OperatorStats):
+    def __add__(self, op_stats: OperatorStats):
```

Comment:
I could be mistaken, but I believe `__iadd__` provides support for the `+=` operator, whereas `__add__` provides support for the `+` operator. Thus, if we want to make this change, we would also need to update every instance of:
```
record_op_stats += other_record_op_stats
```
To be:
```
record_op_stats = record_op_stats + other_record_op_stats
```
Alternatively, we could ignore this linting error in the `ruff.toml` (or perhaps there's a third option which I'm not thinking of?). Overall, I don't have a firm stance on this, other than my belief that support for the `+=` syntax is nice b/c it makes code a tiny bit more readable elsewhere, but I'm happy to switch every `+=` to `+` if you'd prefer. LMK.

Reply:
(I see that you made an update below to the effect of `op_stats = op_stats + other_op_stats`, I think there might be a few places in `operators/` where this change also needs to take effect -- but I'll confirm if we decide to deprecate `+=`.

Reply:
Reverted to `__iadd__` and fixed the issue (was not returning `self`)